### PR TITLE
Add [S3.12] S3 access control lists (ACLs) should not be used to manage user access to buckets. Closes #363

### LINKED ISSUE #363: Add  [S3.12] S3 access control lists (ACLs) should not be used to manage user access to buckets
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Reference**
- https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-fsbp-controls.html#fsbp-s3-12

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by vkumbha**
### Checklist
- [ ] Issue(s) linked

### CODE REVIEW

File: foundational_security/s3.sp [hunk 1]
```diff
@@ -153,4 +153,17 @@ control "foundational_security_s3_11" {
     foundational_security_item_id  = "s3_11"
     foundational_security_category = "logging"
   })
+}
+
+control "foundational_security_s3_12" {
```
Comment (cbruno10) [inline foundational_security/s3.sp:158]: Please add this to the benchmark list above too

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@vkumbha Please see review, thanks!

**Review by cbruno10 — APPROVED** [review]

File: query/s3/s3_bucket_acls_should_prohibit_user_access.sql [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      arn,
+      title,
+      to_jsonb(bucket_acl_permissions) - bucket_owner as additional_permissions,
+      region,
+      account_id 
+   from
+      bucket_acl_details 
+)
+select
+   arn,
```
Comment (misraved) [inline query/s3/s3_bucket_acls_should_prohibit_user_access.sql:30]:
```suggestion
    -- Required Columns
  arn as resource,
```

File: query/s3/s3_bucket_acls_should_prohibit_user_access.sql [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+   case
+      when jsonb_array_length(additional_permissions) = 0 then 'ok' 
+      else 'alarm' 
+   end status, 
+   case
+      when jsonb_array_length(additional_permissions) = 0  then title || ' does not have ACLs for user access.' 
+      else title || ' has ACLs for user access.' 
+   end reason, 
+   region,
```
Comment (misraved) [inline query/s3/s3_bucket_acls_should_prohibit_user_access.sql:39]:
```suggestion
  -- Additional Dimensions
   region, 
```

**Review by misraved — COMMENTED** [review]
Please make changes as per the suggestions.

**Review by rajlearner17 — APPROVED** [review]

**Pull request merged**